web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -515,51 +512,18 @@ export class MapComponent implements AfterViewInit, OnDestroy {
         }
       }
     }
-    if (candidatePolygons.length === 1) {
-      this.zone.run(() => {
-        this.navigationService.selectLocationOfInterest(
-          candidatePolygons[0].get('id')
-        );
-      });
-      return;
-    }
-    this.openSelectLocationOfInterestDialog(candidatePolygons);
-  }
 
-  private openSelectLocationOfInterestDialog(
-    candidatePolygons: google.maps.Polygon[]
-  ) {
-    // TODO: Account for case where polygons in multipolygon overlap.
+    const candidateLois = List(
+      candidatePolygons.map(p => this.lois.find(loi => loi.id === p.get('id')))
```

Comment:
Please extract to a function to improve readability, eg:

```ts
const candidateLois = this.getLoisById(candidateLoiIds);
```

You can also simplify the preceding block with something like:
```ts
const candidateLoiIds = this.getIntersectingLoiIds(event.latLng!);
...
getIntersectingLoiIds(latLng: LatLng): List<String> {
  return this.polygons.values().flatMap(ps => ps.filter(p => containsLocation(latLng p))).map(p => p.get('id'));
}
```